agent/src/agent.ts

```diff
@@ -710,7 +712,10 @@ export class Agent extends MessageHandler {
         })
 
         this.registerAuthenticatedRequest('chat/restore', async ({ modelID, messages, chatID }) => {
-            const chatModel = new SimpleChatModel(modelID, [], chatID)
+            const chatModel =
+                modelID === null
```

Comment:
We should always handle both null and undefined
```suggestion
                modelID === null || modelID === undefined
```

It's normally easier to do do the condition like this
```
const foo = modelID ? ... : /* when null or undefined */
```

Reply:
Note that `modelID?: string` does not allow `null` values, you need to type it as `modelID?: string | null` to make it 100% correct. However, we don't do this consistently everywhere